ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt

```diff
@@ -81,10 +89,16 @@ internal class VariableDataProvider(
         return "$pricePerPeriod ($pricePerMonth/$unit)"
     }
 
-    private fun getIntroPhaseToApply(rcPackage: Package): PricingPhase? {
-        // TODO-PAYWALLS: Decide how this will look like for products that have both a free trial
-        // and a discounted introductory price.
+    private fun getFirstIntroOfferToApply(rcPackage: Package): PricingPhase? {
         val option = rcPackage.product.defaultOption
         return option?.freePhase ?: option?.introPhase
     }
+
+    private fun getSecondIntroOfferToApply(rcPackage: Package): PricingPhase? {
+        val option = rcPackage.product.defaultOption
+        if (option?.freePhase != null) {
```

Comment:
Well, I think this should work as long as google doesn't allow to have an intro price before a free trial, which they currently don't and would also be very weird 😅 . So I think it's ok.

Reply:
Yeah, I thought the same, but I checked and added a intro offer, then a trial, and the trial is still applied first. This is their preview

<img width="772" alt="Screenshot 2023-10-06 at 16 06 14" src="https://github.com/RevenueCat/purchases-android/assets/664544/7bec1fc0-abf1-4966-870a-aee0710a0429">

And yes, that would be strange lol